### PR TITLE
7.0 backport legacy

### PR DESCRIPTION
**Pull request by p-tombez**
Backport modules for legacy support

### CODE REVIEW

File: attachment_swift/README.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
+Attachments on Swift storage
+============================
+
+This addon enable storing attachments (documents and assets) on OpenStack Object Storage (Swift)
```
Comment (asaunier) [inline attachment_swift/README.rst]:
```suggestion
This addon enables storing attachments (documents and assets) on OpenStack Object Storage (Swift).
```

File: base_attachment_object_storage/README.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Base class for attachments on external object store
+===================================================
+
+This is a base addon that regroup common code used by addons targeting specific object store
```
Comment (asaunier) [inline base_attachment_object_storage/README.rst]:
```suggestion
This is a base addon that regroups common code used by addons targeting specific object store.
```

**Review by asaunier — APPROVED** [review]
Same comment than for 8.0 branch.

File: base_attachment_object_storage/models/ir_attachment.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+                    full_path, exc_info=True
+                )
+
+
+@contextmanager
+def savepoint(cursor):
+    name = uuid.uuid1().hex
+    cursor.execute('SAVEPOINT "%s"' % name)
+    try:
+        yield
```
Comment (Tonow-c2c) [inline base_attachment_object_storage/models/ir_attachment.py:46]: we didn't need to add the expression/value to yield after one `yield` ? [https://docs.python.org/3/reference/simple_stmts.html#yield](https://docs.python.org/3/reference/simple_stmts.html#yield)

File: attachment_s3/models/ir_attachment.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                _logger.exception(
+                    'Error during storage of the file %s' % filename
+                )
+                raise except_orm(
+                    _('S3 Error'),
+                    _('The file could not be stored: %s') %
+                    (self._parse_s3_error(error),)
+                )
+        else:
+            _super = super(IrAttachment, self)
```
Comment (Tonow-c2c) [inline attachment_s3/models/ir_attachment.py:137]: mybe add a `TODO: refactor later`

**Review by Tonow-c2c — APPROVED** [review]
It's not a good idea to do like one template global and just change specific things for each version ?
Cause V7 and V8 seems pretty close.

File: base_attachment_object_storage/models/ir_attachment.py (hunk 3 above)
Comment (p-tombez) [reply base_attachment_object_storage/models/ir_attachment.py:46]: It's not mandatory as explained here : https://docs.python.org/3/reference/expressions.html#yieldexpr
But be careful as the "official" use of `yield` is to make generators, in the following doc, you can see how it is used as a context manager (to be use in `with` statements) and a few examples of `yield` usage without value : https://docs.python.org/3/library/contextlib.html

File: attachment_s3/models/ir_attachment.py (hunk 4 above)
Comment (p-tombez) [reply attachment_s3/models/ir_attachment.py:137]: Why ? If I do so, we should also update all other versions as it's the same everywhere

**Comment by p-tombez** [comment]
> It's not a good idea to do like one template global and just change specific things for each version ?
> Cause V7 and V8 seems pretty close.

Yeah, so as v9, 10, 11 and 12. If you want to propose an improvement, go ahead but it's not the goal of this current work

File: base_attachment_object_storage/models/ir_attachment.py (hunk 3 above)
Comment (guewen) [reply base_attachment_object_storage/models/ir_attachment.py:46]: @Tonow-c2c There is no value to yield here. 
About @p-tombez 's remark: the context manager in fact expects and uses a generator that yields only one item.

File: attachment_s3/models/ir_attachment.py (hunk 4 above)
Comment (guewen) [reply attachment_s3/models/ir_attachment.py:137]: There is no reason to change this. It has been done  because of the line length.

**Comment by guewen** [comment]
> Yeah, so as v9, 10, 11 and 12. If you want to propose an improvement, go ahead but it's not the goal of this current work

I agree, except on the "go ahead" part. Yes it is a pity when we have to apply the same fixes on every version, but in any case you would have to test them on every version and then the "small diffs for each version" can be hard to digest. I would add that there have been many tentative to have addons supporting several odoo versions / propositions that odoo support that, and no good outcome ever came out.

**Comment by guewen** [comment]
@p-tombez any special part we need to look at where you needed to change code to adapt to 7.0?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
+[![Build Status](https://travis-ci.com/camptocamp/odoo-cloud-platform.svg?token=Lpp9PcS5on9AGbp76WKB&branch=8.0)](https://travis-ci.com/camptocamp/odoo-cloud-platform)
```
Comment (simahawk) [inline README.md]: branch=7.0

**Review by simahawk — COMMENTED** [review]
@p-tombez ping

**Pull request merged**